### PR TITLE
feat(php): require PHP 7.0

### PR DESCRIPTION
**Pull request by hypeJunction**
BREAKING CHANGE
Elgg now requires PHP 7.0 to run

### CODE REVIEW

**Comment by hypeJunction** [comment]
this should speed up tests a bit

File: engine/tests/classes/Elgg/BaseTestCase.php [hunk 1]
```diff
@@ -170,13 +170,13 @@ protected function tearDown() {
...
 
 		// free up some memory
-		$refl = new \ReflectionObject($this);
-		foreach ($refl->getProperties() as $prop) {
-			if (!$prop->isStatic() && 0 !== strpos($prop->getDeclaringClass()->getName(), 'PHPUnit_')) {
-				$prop->setAccessible(true);
-				$prop->setValue($this, null);
-			}
-		}
+//		$refl = new \ReflectionObject($this);
```
Comment (jdalsem) [inline engine/tests/classes/Elgg/BaseTestCase.php]: why leave this hanging around?

**Comment by jdalsem** [comment]
Fixes #11186

**Comment by jdalsem** [comment]
be advised... upgrading docs have been moved to the appendix

**Comment by mrclay** [comment]
Get rid of `paragonie/random_compat`.

**Comment by mrclay** [comment]
Replaced by #11416

**Pull request closed without merging**